tools/src/main/scala/scala/scalanative/interflow/Intrinsics.scala

```diff
@@ -170,13 +172,36 @@ private[interflow] trait Intrinsics { self: Interflow =>
       case _ if name == arrayLengthIntrinsic =>
         val Seq(arr) = rawArgs
         Some(eval(nir.Op.Arraylength(arr)))
-      case nir.Rt.FromRawPtrSig =>
+      case nir.Global.Member(nir.Rt.Runtime.name, nir.Rt.FromRawPtrSig) =>
         val Seq(_, value) = rawArgs
         Some(eval(nir.Op.Box(nir.Rt.BoxedPtr, value)))
-      case nir.Rt.ToRawPtrSig =>
+      case nir.Global.Member(nir.Rt.Runtime.name, nir.Rt.ToRawPtrSig) =>
         val Seq(_, value) = rawArgs
         Some(eval(nir.Op.Unbox(nir.Rt.BoxedPtr, value)))
+      case nir.Global.Member(nir.Global.Top(className), method) =>
+        method.unmangled match {
+          case nir.Sig.Method(methodName, _, _) =>
+            // Lower.BoxTo and Lower.UnboxTo maps use the companion object name,
+            // but java static methods on the class are also present
+            // and we need to handle them.
+            val isCompanionObject = className.last == '$'
+            val companionObjectName = if (isCompanionObject) {
+              className
+            } else {
+              className + '$'
+            }
+
+            val boxIntrinsic = boxIntrinsicType
+              .get((companionObjectName, methodName))
+              .map(tpe => eval(nir.Op.Box(tpe, rawArgs.last)))
+
+            val unboxIntrinsic = unboxIntrinsicType
+              .get((companionObjectName, methodName))
+              .map(tpe => eval(nir.Op.Unbox(tpe, rawArgs.last)))
```

Comment:
We can make it lazy
```suggestion
            def unboxIntrinsic = unboxIntrinsicType
              .get((companionObjectName, methodName))
              .map(tpe => eval(nir.Op.Unbox(tpe, rawArgs.last)))
```

Reply:
Ended up inlining the two values.
Let me know if you like it.